From: =?utf-8?q?Rapha=C3=ABl_Hertzog?= <hertzog@debian.org>
Date: Fri, 5 Dec 2014 09:36:27 +0100
Subject: Update linux flavours for armhf/armel

With linux 3.16 of Debian Jessie, armmp and armmp-lpae are the
only armhf flavours availale. On armel, we lost the support of
the iop32x flavors.

Linux 3.14 used in Kali doesn't build the ixp4xx flavor, so keep that
one disabled too.
---
 functions/defaults.sh | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/functions/defaults.sh b/functions/defaults.sh
index feebdee..0d5fa4f 100755
--- a/functions/defaults.sh
+++ b/functions/defaults.sh
@@ -525,15 +525,29 @@ Set_defaults ()
 				*)
 					# armel will have special images: one rootfs image and many additional kernel images.
 					# therefore we default to all available armel flavours
-					LB_LINUX_FLAVOURS="${LB_LINUX_FLAVOURS:-iop32x ixp4xx kirkwood orion5x versatile}"
+					case "${LB_DISTRIBUTION}" in
+						wheezy)
+							LB_LINUX_FLAVOURS="${LB_LINUX_FLAVOURS:-iop32x ixp4xx kirkwood orion5x versatile}"
+							;;
+						*)
+							LB_LINUX_FLAVOURS="${LB_LINUX_FLAVOURS:-kirkwood orion5x versatile}"
+							;;
+					esac
 					;;
 			esac
 			;;
 
 		armhf)
 			# armhf will have special images: one rootfs image and many additional kernel images.
-			# therefore we default to all available armel flavours
-			LB_LINUX_FLAVOURS="${LB_LINUX_FLAVOURS:-mx5 omap}"
+			# therefore we default to all available armhf flavours
+			case "${LB_DISTRIBUTION}" in
+				wheezy)
+					LB_LINUX_FLAVOURS="${LB_LINUX_FLAVOURS:-mx5 omap}"
+					;;
+				*)
+					LB_LINUX_FLAVOURS="${LB_LINUX_FLAVOURS:-armmp armmp-lpae}"
+					;;
+			esac
 			;;
 
 		amd64)
